src/server/pps/server/api_server.go

```diff
@@ -712,6 +712,37 @@ func (a *apiServer) ListJob(ctx context.Context, request *ppsclient.ListJobReque
 	}, nil
 }
 
+func (a *apiServer) DeleteJob(ctx context.Context, request *ppsclient.DeleteJobRequest) (response *google_protobuf.Empty, err error) {
+	func() { a.Log(request, nil, nil, 0) }()
+	metricsFn := metrics.ReportUserAction(ctx, a.reporter, "DeleteJob")
+	defer func(start time.Time) { metricsFn(start, err) }(time.Now())
+
+	pfsAPIClient, err := a.getPfsClient()
+	if err != nil {
+		return nil, err
+	}
+	persistClient, err := a.getPersistClient()
+	if err != nil {
+		return nil, err
+	}
+	jobInfo, err := persistClient.InspectJob(ctx, &ppsclient.InspectJobRequest{
+		Job: request.Job,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err := a.deleteJob(ctx, jobInfo); err != nil {
+		return nil, err
+	}
+	if _, err := pfsAPIClient.DeleteRepo(ctx, &pfsclient.DeleteRepoRequest{Repo: jobInfo.OutputCommit.Repo}); err != nil {
+		return nil, err
+	}
```

Comment:
Yes! That would be unfriendly to say the least.

In that vein ...

Do we want to allow someone to delete a job that belongs to a pipeline? Under what conditions is that helpful? What happens if you delete a job belonging to a pipeline while its running?

Wouldn't the pipeline reschedule the job? That might be an interesting way to force a job to re-run (if its laggy ... or perhaps you scaled up your VMs because it was being slow and want to run the job again so that its distributed across more VMs)